design-documents/graph-ql/coverage/url-resolver-identifier.md

```diff
@@ -0,0 +1,109 @@
+ # UrlResolver identifier 
+ ## Problem statement   
+ 
+In the client flow, for PWA, in order to load a entity/template, one must make a query to the `urlResolver` to obtain the *id* and *Entity Type* of the content that can be Product, Category or CMS.
+
+#### Example:
+##### Request:
+```graphql
+urlResolver(url: "/breathe-easy-tank") {
+  id
+  type
+}
+```
+##### Response:
+```json
+{
+  "data": {
+    "urlResolver": {
+      "id": 1820,
+      "type": "PRODUCT"
+    }
+  }
+}
+```
+
+Currently we can't use this Id in Product to filter by, because of how Search API works, also CMS has another unique identifier that can be used.
+Also we want not to expose database autoincrement ID in graphql in the future because of replacements with UUID.
+
+ ## Proposed solution
+
+Introduction of `urlResolverV2` and deprecation of `urlResolver`. We want to keep the same id name, but change the type, and do this in a backward compatible way. So a new query is needed.
+
+```graphql
+urlResolverV2(url: String) {
+  id: ID!
+  ...
+}
+```
+
+This ID has to be filterable in all entities, even in product
+
+ ```graphql
+ products(filter: { id: { eq: "$valueOfIdentifierFromUrlResolver"}}) {
```

Comment:
Why do we need to support `eq` and not just `in`?

Reply:
removed